website/src/docs/hotchocolate/security/authorization.md

```diff
@@ -0,0 +1,356 @@
+
+If the `@authorize` directive is specified on a type, it is applied to all fields of that type. Specified on an individual field the directive precedences the one on the type.
+
+<ExampleTabs>
+<ExampleTabs.Annotation>
+
+In the Annotation-based approach we can use the `[Authorize]` attribute to add the `@authorize` directive.
+
+```csharp
+[Authorize]
+public class User
+{
+    public string Name { get; set; }
+
+    [Authorize]
+    public Address Address { get; set; }
+}
+```
+
+> ⚠️ Note: We need to use the `HotChocolate.AspNetCore.AuthorizationAttribute` instead of the `Microsoft.AspNetCore.AuthorizationAttribute`.
+
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
+
+```csharp
+public class UserType : ObjectType<User>
+{
+    protected override void Configure(IObjectTypeDescriptor<User> descriptor)
+    {
+        descriptor.Authorize();
+
+        descriptor.Field(f => f.Address).Authorize();
+    }
+}
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+```sdl
+type User @authorize {
+  name: String!
+  address: Address! @authorize
+}
+```
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+If we do not specify any arguments to the `@authorize` directive, it will only enforce that the requestor is authenticated, nothing more. If he is not and he tries to access an authorized field, a GraphQL error will be raised and the field result set to `null`.
+
+## Roles
+
+Roles provide a very intuitive way of dividing our users into groups with different access rights.
+
+When building our `ClaimsPrincipal`, we just have to add one or more role claims.
+
+```csharp
+claims.Add(new Claim(ClaimTypes.Role, "Administrator"));
+```
+
+We can then check, whether an authenticated user has these role claims.
+
+<ExampleTabs>
+<ExampleTabs.Annotation>
+
+```csharp
+[Authorize(Roles = new [] { "Guest", "Administrator" })]
+public class User
+{
+    public string Name { get; set; }
+
+    [Authorize(Roles = new[] { "Administrator" })]
+    public Address Address { get; set; }
+}
+```
+
+</ExampleTabs.Annotation>
+<ExampleTabs.Code>
+
+```csharp
+public class UserType : ObjectType<User>
+{
+    protected override Configure(IObjectTypeDescriptor<User> descriptor)
+    {
+        descriptor.Authorize(new[] { "Guest", "Administrator" });
+
+        descriptor.Field(t => t.Address).Authorize(new[] { "Administrator" });
+    }
+}
+```
+
+</ExampleTabs.Code>
+<ExampleTabs.Schema>
+
+```sdl
+type User @authorize(roles: [ "Guest", "Administrator" ]) {
+  name: String!
+  address: Address! @authorize(roles: "Administrator")
+}
+```
+
+</ExampleTabs.Schema>
+</ExampleTabs>
+
+> ⚠️ Note: If multiple roles are specified, a user only has to match one of the specified roles, in order to be able to execute the resolver.
+
+[Learn more about role-based authorization in ASP.NET Core](https://docs.microsoft.com/aspnet/core/security/authorization/roles)
+
+## Policies
```

Comment:
It's probably worth adding a similar link for policies to the asp.net core docs as done just above for roles.

Reply:
I had one initially, must've gotten lost, while I was shifting stuff around. Good catch!